File: library_base/src/main/java/com/hjc/library_base/adapter/BaseRecyclerAdapter.java
package com.hjc.library_base.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;

/**
 * @Author: HJC
 * @Date: 2019/1/4 14:52
 * @Description: RecyclerView Adapter基类
 */
public abstract class BaseRecyclerAdapter extends RecyclerView.Adapter<BaseRecyclerAdapter.MyViewHolder> {

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ViewDataBinding viewDataBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), getLayoutId(), parent, false);
        return new MyViewHolder(viewDataBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {

    }

    public abstract int getLayoutId();

    static class MyViewHolder extends RecyclerView.ViewHolder {
        final ViewDataBinding mBinding;

        MyViewHolder(ViewDataBinding binding) {
            super(binding.getRoot());
            this.mBinding = binding;
        }
    }

}

File: library-common/src/main/java/com/hjc/library_common/adapter/BaseRecyclerAdapter.kt
package com.hjc.library_common.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView

/**
 * @Author: HJC
 * @Date: 2019/1/4 14:52
 * @Description: RecyclerView Adapter基类
 */
abstract class BaseRecyclerAdapter : RecyclerView.Adapter<BaseRecyclerAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val viewDataBinding = DataBindingUtil.inflate<ViewDataBinding>(
            LayoutInflater.from(parent.context),
            getLayoutId(),
            parent,
            false
        )
        return MyViewHolder(viewDataBinding)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {

    }

    abstract fun getLayoutId(): Int

    class MyViewHolder(mBinding: ViewDataBinding) : RecyclerView.ViewHolder(
        mBinding.root
    )
}